ampd/src/asyncutil/future.rs

```diff
@@ -1,102 +1,45 @@
-use std::pin::Pin;
-use std::task::{Context, Poll};
 use std::time::Duration;
 
-use futures::{Future, FutureExt};
+use futures::Future;
 use tokio::time;
 
-pub fn with_retry<F, Fut, R, Err>(
-    future: F,
-    policy: RetryPolicy,
-) -> impl Future<Output = Result<R, Err>>
+pub async fn with_retry<F, Fut, R, Err>(mut future: F, policy: RetryPolicy) -> Result<R, Err>
```

Comment:
All good, I'm wondering what are the differences these 2 different implementations have?